### ISSUE TITLE
Documentation for other-modules

### ISSUE THREAD

**Issue by harendra-kumar** [haskell/haddock#870]
Is there an option or a way to generate documentation for `other-modules` in the cabal file? I want to keep some modules hidden from users but I would like to have documentation for those for developers. How can I do that? I looked at the docs but could not figure out if there is way to do this.

I tried using a flag in cabal file like this:

```
library
    hs-source-dirs:    src
    if flag(dev)
        exposed-modules:
                       Streamly.SVar
                     , Streamly.Streams.StreamK
                     , Streamly.Streams.StreamD
                     , Streamly.Streams.Serial
                     , Streamly.Streams.SVar
                     , Streamly.Streams.Async
                     , Streamly.Streams.Parallel
                     , Streamly.Streams.Ahead
                     , Streamly.Streams.Zip
                     , Streamly.Streams.Prelude
    else
        other-modules:
                       Streamly.SVar
                     , Streamly.Streams.StreamK
                     , Streamly.Streams.StreamD
                     , Streamly.Streams.Serial
                     , Streamly.Streams.SVar
                     , Streamly.Streams.Async
                     , Streamly.Streams.Parallel
                     , Streamly.Streams.Ahead
                     , Streamly.Streams.Zip
                     , Streamly.Streams.Prelude
```

But `cabal sdist` does not like this and warns that this won't be accepted by hackage. Maybe this can be fixed in cabal but it would be nice if haddock has an option include the documentation for `other-module` that way we won't need a flag for this purpose.

**Comment by harpocrates** [comment]
Haddock does have such an option already: it is `--show-all`. I'm not sure what the best way is to thread it to `cabal` beyond just `--haddock-options`.

## Example

Take `bytestring` as an example (it has a bunch of `other-modules`):

     $ git clone https://github.com/haskell/bytestring && cd bytestring

Here is the index after generating plain old Haddock's

     $ cabal new-haddock --with-ghc /usr/local/bin/ghc-8.4 \
                         --with-haddock /usr/local/bin/haddock-ghc-8.4.2 \
                         --allow-newer
     $ open dist-newstyle/build/x86_64-osx/ghc-8.4.2/bytestring-0.10.9.0/doc/html/bytestring/index.html

<img width="597" alt="screen shot 2018-07-01 at 2 59 40 pm" src="https://user-images.githubusercontent.com/10766081/42137703-6b8a9794-7d3f-11e8-869b-23510e9a43f2.png">

Now with `--show-all`:

     $ cabal new-haddock --with-ghc /usr/local/bin/ghc-8.4 \
                         --with-haddock /usr/local/bin/haddock-ghc-8.4.2 \
                         --allow-newer \
                         --haddock-options '--show-all'
     $ open dist-newstyle/build/x86_64-osx/ghc-8.4.2/bytestring-0.10.9.0/doc/html/bytestring/index.html

<img width="629" alt="screen shot 2018-07-01 at 3 00 26 pm" src="https://user-images.githubusercontent.com/10766081/42137718-8c892226-7d3f-11e8-882b-06c7f1e5ed4f.png">

**Comment by harendra-kumar** [comment]
@harpocrates that's great, thanks, I will try it out. 

But it does not seem to be documented in this guide - http://haskell-haddock.readthedocs.io/en/latest/invoking.html nor in this one https://www.haskell.org/haddock/doc/html/invoking.html . Is this something new not documented? Or I am not looking at the right place?

**Comment by harpocrates** [comment]
Fixed in a631fb6c15de83042e9739285518fda72cfb3144.
